torch/_dynamo/variables/builtin.py

```diff
@@ -1176,6 +1177,10 @@ def _grad_changed(old, new):
                 return obj.var_getattr(tx, name).clone(source=source)
             except NotImplementedError:
                 return GetAttrVariable(obj, name, **options)
+        elif isinstance(obj, TorchInGraphFunctionVariable):
+            member = getattr(obj.value, name)
+            if trace_rules.lookup(member) is not None:
+                return trace_rules.lookup(member)(member, **options)
         elif isinstance(obj, TorchVariable):
             member = getattr(obj.value, name)
             if is_utils_checkpoint(member):
```

Comment:
What's going on with the checkpoint special case here? Is TorchInGraphFunctionVariable guaranteed to not have checkpoint?

Reply:
Yes, ```torch.utils.checkpoint.checkpoint``` is a skipped function, not in graph one.